atc/exec/task_step.go

```diff
@@ -185,7 +183,7 @@ func (step *TaskStep) Run(ctx context.Context, state RunState) error {
 		return err
 	}
 
-	processSpec := worker.TaskProcessSpec{
+	processSpec := runtime.ProcessSpec{
```

Comment:
Are we still in Core? Would it be better to have a `func NewProcessSpec()` instead of exposing the `struct` here?

Reply:
core is allowed to consume any method or struct in the runtime package. Is there a benefit to exposing the structs through constructors or factories. I couldn't see any, anyone have opinions/ideas on this?

Reply:
Bin and I chatted a bit offline about this - this might be a Philosophical Question about how we use Go lol. we typically have constructors that return pointers to private structs in other places of the code, but I _believe_ that's to add more control around how packages interact. In the case here, where we're explicitly setting up this package just to hold types that are designed to be sharable, I think there is less need to have a constructor